### PR TITLE
add loongson endian check

### PR DESCRIPTION
**Pull request by loongson-zjl**
add loongarch endian check . reference #53

### CODE REVIEW

**Review by xen0n — COMMENTED** [review]
Maybe put the code block after ARM and before MIPS to keep the local blocks alphabetically sorted?

Also AFAIK the architecture is spelled "LoongArch" canonically, so please adjust. As you can see, the "PowerPC" is spelled with canonical cases so you don't have to all-caps the LoongArch name.

**Comment by loongson-zjl** [comment]
> Maybe put the code block after ARM and before MIPS to keep the local blocks alphabetically sorted?
> 
> Also AFAIK the architecture is spelled "LoongArch" canonically, so please adjust. As you can see, the "PowerPC" is spelled with canonical cases so you don't have to all-caps the LoongArch name.

I have changed

File: src/sph_types.h [hunk 1]
```diff
@@ -1057,6 +1057,13 @@ typedef long long sph_s64;
 
 #define SPH_DETECT_LITTLE_ENDIAN     1
 
+/*
+ * LoongArch, little-endian.
```
Comment (xen0n) [inline src/sph_types.h]: Sorry, I missed this comment. All LoongArch systems are little-endian per the manual, so this line should simply read:

```suggestion
 * LoongArch.
```

Looking at the context around the "MIPS, little-endian" block you can find an adjacent block for "MIPS, big-endian", so LoongArch doesn't need the "little-endian" qualifier.

**Review by xen0n — COMMENTED** [review]
Additionally, please squash your changes into one commit after making this one change. This is good for reducing pollution in the version control history.

**Pull request closed without merging**